### PR TITLE
GHA - refactor versioning - [MOD-5732]

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

This PR introduces the `event-release.yml` workflow, which moves version snapshots on S3 buckets and auto-generates a version-bumping PR.
It also implements a caching mechanism for snapshot generation, to skip the nightly build if no changes were introduced on this day.

### CODE REVIEW

File: .github/workflows/event-release.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          LINE: '#define REDISEARCH_VERSION_PATCH'
+        run: sed -i "s/^${{ env.LINE }} [0-9]\+$/${{ env.LINE }} ${{ steps.verify.outputs.next_patch }}/" src/version.h
+
+      - name: Commit and push
+        run: |
+          git config --global user.name "${{ github.triggering_actor }}"
+          git config --global user.email "${{ github.event.sender.id }}+${{ github.event.sender.login }}@users.noreply.github.com"
+          git checkout -b bump-version-${{ steps.verify.outputs.next_version }}
+          git add src/version.h
+          git commit -m "Bump version from $cur_version to $next_version"
```
Comment (GuyAv46) [inline .github/workflows/event-release.yml:42]: Fix variables

File: .github/workflows/event-release.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          VERSION: ${{ needs.update-version.outputs.next_version }}
+          VER_BRANCH: ${{ github.event.release.target_commitish }}
+        run: |
+          for DIR in {$ENT_DIR,$OSS_DIR}; do
+            ARTIFACTS=$(aws s3 ls $DIR/snapshots/redisearch | grep '.${{ env.VER_BRANCH }}.zip$' | awk '{print $NF}')
+            echo "Handling artifacts:"
+            echo "$ARTIFACTS"
+            echo "----------------- END -----------------"
+            for ARTIFACT in $ARTIFACTS; do
+              echo -n "Setting $DIR/snapshots/$ARTIFACT to $DIR/${ARTIFACT/.${{ env.VER_BRANCH }}.zip/.${{ env.VERSION }}.zip} ... "
```
Comment (GuyAv46) [inline .github/workflows/event-release.yml:88]: Use variables directly

File: .github/workflows/event-release.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          import os
+          with open(os.environ["GITHUB_OUTPUT"], "a") as fp:
+            print(f"cur_version={major}.{minor}.{patch}", file=fp)
+            print(f"next_version={major}.{minor}.{patch+1}", file=fp)
+            print(f"next_patch={patch+1}", file=fp)
+
+      - name: Update version for next patch
+        env:
+          LINE: '#define REDISEARCH_VERSION_PATCH'
+        run: sed -i "s/^${{ env.LINE }} [0-9]\+$/${{ env.LINE }} ${{ steps.verify.outputs.next_patch }}/" src/version.h
```
Comment (alonre24) [inline .github/workflows/event-release.yml:34]: explain this

File: .github/workflows/flow-build-artifacts.yml [hunk 4]
```diff
@@ -45,12 +45,17 @@ on:
       redis-reference:
         type: string
 
+env:
+  SHA_CACHE_NAME: ${{ inputs.reference || github.ref_name }}-${{ github.event_name }}-last-sha
```
[the remark this answers is not in the document]
Comment (alonre24) [reply .github/workflows/flow-build-artifacts.yml:49]: and explain shortly the key -> path -> content mechanism

**Review by alonre24 — APPROVED** [review]
Good comments added, great job!

**Pull request merged**